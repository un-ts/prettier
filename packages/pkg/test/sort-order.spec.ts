import { format } from 'prettier'

import pkg2 from './fixtures/fixture2.json'

import PkgPlugin from 'prettier-plugin-pkg'

test('sort-order', async () => {
  const input = JSON.stringify(pkg2, null, 2)
  const output = await format(input, {
    filepath: 'package.json',
    parser: 'json-stringify',
    plugins: [PkgPlugin],
    packageSortOrder: ['name', 'version', 'description', 'scripts'],
  })

  expect(output).toMatchSnapshot()
})
