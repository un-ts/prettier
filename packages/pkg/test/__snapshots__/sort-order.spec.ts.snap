// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sort-order 1`] = `
"{
  "name": "prettier-plugin-pkg",
  "version": "0.0.0",
  "description": "An opinionated package.json formatter plugin for Prettier",
  "scripts": {
    "bootstrap": "yarn build && tsc --module commonjs --outDir lib/cjs",
    "build": "ts-node -O '{\\"module\\":\\"commonjs\\"}' build"
  },
  "repository": "git@github.com/un-ts/prettier.git",
  "homepage": "https://github.com/un-ts/prettier/tree/master/packages/pkg",
  "author": "JounQin <admin@1stg.me> (https://www.1stG.me)",
  "license": "MPL-2.0",
  "engines": {
    "node": ">= 8.0.0"
  },
  "main": "lib/cjs",
  "module": "lib/esm",
  "es2015": "lib/es2015",
  "files": [
    "lib",
    "rules",
    "README.md",
    "LICENSE"
  ],
  "keywords": [
    "package",
    "package.json",
    "pkg",
    "plugin",
    "prettier",
    "prettier-plugin"
  ],
  "peerDependencies": {
    "prettier": "^1.18.2"
  }
}
"
`;
